core/src/main/java/hudson/Util.java

```diff
@@ -1432,36 +1432,76 @@ public static Number tryParseNumber(@CheckForNull String numberStr, @CheckForNul
     }
 
     /**
-     * Checks if the method defined on the base type with the given arguments
-     * is overridden in the given derived type.
-     */
-    public static boolean isOverridden(@NonNull Class base, @NonNull Class derived, @NonNull String methodName, @NonNull Class... types) {
-        return !getMethod(base, methodName, types).equals(getMethod(derived, methodName, types));
+     * Checks whether the method defined on the base type with the given arguments is overridden in the given derived
+     * type.
+     *
+     * @param base       The base type.
+     * @param derived    The derived type.
+     * @param methodName The name of the method.
+     * @param types      The types of the arguments for the method.
+     * @return {@code true} when {@code derived} provides the specified method other than as inherited from {@code base}.
+     */
+    public static boolean isOverridden(@NonNull Class<?> base, @NonNull Class<?> derived, @NonNull String methodName, @NonNull Class<?>... types) {
+        // If derived is not a subclass or implementor of base, it can't override any method
+        // Technically this should also be triggered when base == derived, because it can't override its own method, but
+        // the unit tests explicitly test for that as working.
+        if (!base.isAssignableFrom(derived)) {
+            throw new IllegalArgumentException("The specified derived class (" + derived.getCanonicalName() + ") does not derive from the specified base class (" + base.getCanonicalName() + ").");
+        }
+        final Method baseMethod = Util.getMethod(base, null, methodName, types);
+        if (baseMethod == null) {
```

Comment:
Should we not also check that `baseMethod.getDeclaringClass() == base`?

Reply:
The method sadly did not document as a reuirement, and is expected to work even when the class given does not itself define the method.
An org search seemed to turn up at least one case where the method is called to check whether a class overrides `equals`, `hashCode` and `toString`, where `base` is _not_ passed as `Object.class`, and also does not itself override those methods (https://github.com/jenkinsci/git-plugin/blob/c99d18fcbd064bb9c4a4c2d8ec5ce28217042846/src/test/java/jenkins/plugins/git/traits/GitSCMExtensionTraitTest.java#L34). So adding the restriction would seem to cause a regression there.

Reply:
Well, that is just test code that could be fixed at any time, but point taken that there _could_ be production plugin code floating around somewhere (outside @jenkinsci) that improperly passes a subtype of the type defining the method for some reason.